pkg/resourceinterpreter/configurableinterpreter/configmanager/manager.go

```diff
@@ -63,17 +64,21 @@ func (configManager *interpreterConfigManager) HasSynced() bool {
 // the configurations in the cache.
 func NewInterpreterConfigManager(inform genericmanager.SingleClusterInformerManager) ConfigManager {
 	manager := &interpreterConfigManager{
-		lister:        inform.Lister(resourceInterpreterCustomizationsGVR),
 		initialSynced: &atomic.Value{},
 		configuration: &atomic.Value{},
 	}
 	manager.configuration.Store(make(map[schema.GroupVersionKind]CustomAccessor))
 	manager.initialSynced.Store(false)
-	configHandlers := fedinformer.NewHandlerOnEvents(
-		func(_ interface{}) { manager.updateConfiguration() },
-		func(_, _ interface{}) { manager.updateConfiguration() },
-		func(_ interface{}) { manager.updateConfiguration() })
-	inform.ForResource(resourceInterpreterCustomizationsGVR, configHandlers)
+
+	if inform != nil {
```

Comment:
When will the `inform` be nil?

By the way, `inform` should be `informer`, it's a typo brought by the previous patch.

Reply:
> When will the `inform` be nil?
In executing mode, config is not synced from informer, it's loaded by
https://github.com/karmada-io/karmada/blob/47cb8a683eda20c5cd6cd0e7db57f0280667f7f9/pkg/karmadactl/interpret/execute.go#L82

So informer is unused in this mode.



> By the way, `inform` should be `informer`, it's a typo brought by the previous patch.

updated

Reply:
Have you met a case that the `informer` is `nil`?

I'm still confused about why to add the `if inform != nil`.

Reply:
In interpret command, we don't use informer, so it's nil.

Reply:
Maybe we can add some comments to explain the situation where informer is nil.

Reply:
> Maybe we can add some comments to explain the situation where informer is nil.

Updated.